mjbramich/kandoo
# Column interactions

## Description

Enhancements and Refactoring for Column Management on the board,
Better optimisation of state and  state structure.

### New Column Creation:
Implemented the ability for users to create new columns within the application.
Sent column creation data to the backend for persistence, ensuring newly created columns now reside in the database.
Enhanced the createColumn controller function to respond with the newly created column

### Deletion of a Column:
Implemented the ability for users to delete existing columns within the application.
The selected column to delete is deleted in our db for persistence, ensuring columns are up to date in the db.

### Updating a Columm:
User can now update column title,
Updated Data is sent to the backendin the db for data persistance.

### Fixing of Routes
Column Routes in server was imported as tasks, fixed this.


## Related Issue

-   [ ] Is this related to a specific issue? Is so please specify:

## Type of Changes

Use a check for ✓ the following type of changes:

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
| ✓    | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before

[If applicable, include a picture of the behavior before the changes were made. I.e. a screenshot of the behavior before the changes were made.]

### After

[I.e. a screenshot of the behavior after the changes were made.]

## Testing Steps / QA Criteria

Now able to add, delete, update columns with the ui.
To check we are changing our state correctly please use React dev tools and look at the board context state after each action.

## Checklist:
